src/controllers/bus.js

```diff
@@ -0,0 +1,82 @@
+import { paginate } from 'paginate-info';
+import successRes from '../helpers/successHandler';
+import errorRes from '../helpers/errorHandler';
+import model from '../database/models';
+
+const { Bus } = model;
+export const createBus = async (req, res) => {
+  try {
+    const bus = await Bus.create(req.body);
+    if (!bus) {
+      return errorRes(res, 404, res.__('There was an error while creating a bus'));
+    }
+    return successRes(res, 201, res.__('Bus created successfully'), bus);
+  } catch (error) {
+    return errorRes(res, 500, error);
+  }
+};
+export const getBus = async (req, res) => {
+  try {
+    const { query: { page = 1, limit = 10 } } = req;
+    const offset = (page - 1) * limit;
+    const { rows, count } = await Bus.findAndCountAll({
+      page,
+      limit,
+      offset,
+    });
+    const pagination = paginate(page, count, rows, limit);
+
+    if (offset >= count) {
+      errorRes(res, 401, res.__('There are no buses registered in the system'));
+    }
+    return successRes(res, 200, pagination, rows);
+  } catch (error) {
+    return errorRes(res, 500, `Internal server error : ${error.message}`);
+  }
+};
+export const getOneBus = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const bus = await Bus.findOne({ where: { id } });
+    if (!bus) {
+      return errorRes(res, 404, res.__('Bus not found'));
+    }
+    return successRes(res, 200, res.__('Bus found'), bus);
+  } catch (error) {
+    return errorRes(res, 500, `Internal server error : ${error.message}`);
+  }
+};
+export const updateBus = async (req, res) => {
+  try {
+    const busID = req.params;
```

Comment:
Nice work marius, you can destruct `busID` from `req.params` to avoid future confusions or errors.

Reply:
could you work on this